arangod/Cluster/ClusterInfo.cpp

```diff
@@ -546,10 +547,43 @@ DECLARE_HISTOGRAM(arangodb_load_current_runtime, ClusterInfoScale,
 DECLARE_HISTOGRAM(arangodb_load_plan_runtime, ClusterInfoScale,
                   "Plan loading runtimes [ms]");
 
+DECLARE_GAUGE(arangodb_internal_cluster_info_memory, std::size_t,
+              "Memory cluster info");
+
+namespace {
+struct CountingMemoryResource : std::pmr::memory_resource {
+  explicit CountingMemoryResource(memory_resource* base,
+                                  metrics::Gauge<std::size_t>& metric)
+      : base(base), metric(metric) {}
+
+ private:
+  void* do_allocate(size_t bytes, size_t alignment) override {
+    metric += bytes;
+    return base->allocate(bytes, alignment);
```

Comment:
Since `memory_resource::allocate` may throw, we have to change the order
```suggestion
    auto* p = base->allocate(bytes, alignment);
    metric += bytes;
    return p;
```

Reply:
Thanks! Good point!